docs/12.EXT-REFERENCE-MODULE.md

```diff
@@ -0,0 +1,182 @@
+#  Module API
+
+**Note**: This extension depends on compiler support because it uses `__attribute__` extensions provided only by GCC and
+LLVM/clang. For other compilers, no support has been added yet.
```

Comment:
Actually, I just realized that only the JerryScript modules will be affected. If you create modules using some other mechanism and provide a resolver, that will work. So, I'll document *that*.